components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -112,21 +386,27 @@ void SHOCMacrophysics::set_required_field_impl (const Field<const Real>& f) {
   // at the beginning of the run call. Other than that, there would be really
   // no need to store a scream field here; we could simply set the view ptr
   // in the Homme's view, and be done with it.
-  m_shoc_fields_in.emplace(f.get_header().get_identifier().name(),f);
+  const auto& name = f.get_header().get_identifier().name();
+  m_shoc_fields_in.emplace(name,f);
+  m_shoc_host_views_in[name] = Kokkos::create_mirror_view(f.get_view());
```

Comment:
If you really need host views, do not create the mirror manually. The stored view might not be what you think for subviewed fields (like tracers). Instead, do (for a scalar 3d field)

`auto view_h = my_field.get_reshaped_view<Real**, Host>();`

The method `get_view` is too risky now (I can give you more details offline if you want), and I will likely remove it (or make it private, accessible only to some friends).

Reply:
I believe these are also unnecessary, since I never access host_views anywhere.